### PR TITLE
feat!: 2.1 Components

### PR DESCRIPTION
**Pull request by Jaryt**
Implementing the 2.1 components seen in this abstract component graph:

image

Orbs will not be included with this PR

### CODE REVIEW

**Comment by Jaryt** [comment]
Implements #60, #61, and #60

File: src/lib/Components/Commands/Reusable/index.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    parameters?: CustomParametersList<CommandParameterLiteral>,
+    steps?: Command[],
+  ) {
+    super();
+    this.name = name;
+    this.parameters = parameters || new CustomParametersList();
+    this.steps = steps || [];
+  }
+
+  generate(): unknown {
```
Comment (KyleTryon) [inline src/lib/Components/Commands/Reusable/index.ts:29]: Increase specificity

File: src/lib/Components/Executor/ReusableExecutor.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    this.parameters = parameters || new CustomParametersList();
+  }
+  /**
+   * Generate Reusable Executor schema.
+   * @returns The generated JSON for the Reusable Executor.
+   */
+  generate(): ReusableExecutorSchema {
+    return {
+      executor: {
+        name: this.name,
```
Comment (KyleTryon) [inline src/lib/Components/Executor/ReusableExecutor.ts:47]: Parameters missing

File: src/lib/Components/Job/index.ts [hunk 3]
```diff
@@ -26,27 +27,36 @@ export class Job extends Component {
...
-      ...generatedExecutor,
-    };
+
+    return { steps: generatedSteps, ...generatedExecutor };
+  }
+  /**
+   * Generate Job schema
+   * @returns The generated JSON for the Job.
+   */
+  generate(): unknown {
```
Comment (KyleTryon) [inline src/lib/Components/Job/index.ts:57]: Schema type should be known

File: src/lib/Components/Workflow/index.ts [hunk 4]
```diff
@@ -23,11 +23,16 @@ export class Workflow {
...
    * @param jobs - A list of jobs to be executed as part of your Workflow.
    */
-  constructor(name: string, jobs?: Job[]) {
+  constructor(name: string, jobs?: Array<Job | WorkflowJob>) {
     this.name = name;
-    jobs?.forEach((job) => {
-      this.jobs.push(new WorkflowJob(job));
-    });
+
+    if (jobs?.find((job: unknown) => job instanceof Job)) {
```
Comment (KyleTryon) [inline src/lib/Components/Workflow/index.ts:29]: Remove unknown

File: tsconfig.json [hunk 5]
```diff
@@ -69,7 +69,7 @@
     "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */,
     "resolveJsonModule": true
   },
-  "include": ["src/**/*"],
+  "include": ["src/**/*", "tests"],
```
Comment (KyleTryon) [inline tsconfig.json:72]: why was tests added. We dont need to compile tests

**Comment by eddiewebb** [comment]
Pretty please.

**Comment by Jaryt** [comment]
Intentionally leaving the tests as they are. Ready to be merged into main.

**Review by KyleTryon — APPROVED** [review]

**Pull request merged**